File: app/src/main/java/com/tooploox/aimtask/TracksListDataSource.java
package com.tooploox.aimtask;

interface TracksListDataSource {

    interface TrackView {

        void showTrackArtist(String artist);

        void showTrackTitle(String title);

        void showTrackLength(String length);

        void showTrackThumbnail(String thumbnailUrl);

        void showIsPlayingNow(boolean isPlayingNow);
    }

    int getItemCount();

    void bindViewHolder(TrackView trackView, int position);
}

File: app/src/main/java/com/tooploox/aimtask/TracksListDataSource.kt
package com.tooploox.aimtask

interface TracksListDataSource {

    val itemCount: Int

    interface TrackView {

        fun showTrackArtist(artist: String)

        fun showTrackTitle(title: String)

        fun showTrackLength(length: String)

        fun showTrackThumbnail(thumbnailUrl: String)

        fun showIsPlayingNow(isPlayingNow: Boolean)
    }

    fun bindViewHolder(trackView: TrackView, position: Int)
}
